wagit/sources/Repository.cpp

```diff
@@ -0,0 +1,75 @@
+//
+// Created by Ranveer Aggarwal on 27/08/20.
+//
+
+#include "../headers/Repository.h"
+
+namespace WAGit
+{
+    Repository::Repository(const Path &path, bool force)
+    {
+        this->workTree = path;
+        this->gitDirectory = path / ".git";
+
+        if (!(force || FS::is_directory(this->gitDirectory)))
+            throw Exceptions::NotAGitRepositoryException(this->gitDirectory.u8string());
+
+        Path configFile = GetRepoPath({"config"});
+        if (!force && !FS::exists(configFile))
+            throw Exceptions::GeneralException("Configuration file missing");
+
+        this->config.InitConfig(configFile);
+
+        if (!force && this->config.GetVersion() != 0)
```

Comment:
Should this be ==?

Reply:
Why is this !=?

Reply:
It's kind of git compatibility version. We only support 0 for now. 1 is harder to implement. 
More on that here: https://stackoverflow.com/a/33464086/2302611